### ISSUE TITLE
Feature request: Custom locators

### ISSUE THREAD

**Issue by Phault** [moonrepo/proto#433]
Basically I'd like to be able to run:

```shell
proto plugin add git-locator "github:cool-guy/git-locator-plugin"

# `git` locator is now usable:
proto plugin add my-plugin "git:example.com/cool-guy/my-plugin.git"

# maybe even allow multiple locators from the same locator plugin:
# proto plugin add my-plugin3 "git+https:example.com/cool-guy/my-plugin.git"
# proto plugin add my-plugin2 "git+ssh:git@example.com:cool-guy/my-plugin.git"

proto pin my-plugin@latest --resolve
```

Which would update the .prototools like so:

```toml
my-plugin = "1.0.0"

[plugins]
git-locator = "github:cool-guy/git-locator-plugin"
my-plugin = "git:example.com/cool-guy/my-plugin.git"
```

This will of course complicate the installation process a little, as it would first have to install the locators, before it can handle the usages of them. Essentially I think it should install all plugins using locators it recognises in a loop, until it has installed everything possible. If there's still unknown locators in use, then complain to the user.

There's also the issue of what if two plugins provide locators with the same name? First entry in list wins? I'm not sure.

Aside from providing locators like the git locator in #431, which should probably be bundled anyway, this would also make it possible to create a registry plugin which could resolve from something similar to https://github.com/asdf-vm/asdf-plugins.

Alternatively locators could be separate from plugins, perhaps even called providers or registries, seeing as they're quite a bit different from the regular plugins.

**Comment by milesj** [comment]
I don't think this will ever happen, as we would lose all type safety on the rust side. The `PluginLocator` type (https://docs.rs/warpgate/0.11.6/warpgate/enum.PluginLocator.html) is used heavily throughout the codebase, including moons, and I would rather not lose that.

**Comment by Phault** [comment]
Ah, that's a shame but understandable.

Is there any official plans for registry support down the line?

**Comment by W1M0R** [comment]
It looks like wapm.io now redirects to wasmer.io: https://wasmer.io/
